<launch>
    <!-- <include file="$(find legged_config)/launch/leg_bringup.launch">
        <arg name="rviz" value="false"/>
    </include> -->

     <node pkg="tf" type="static_transform_publisher" name="world_to_odom_broadcaster"
        args="0 0 0 0 0 0 world odom 50" />

  <!-- size of map, change the size in x, y, z according to your application -->
  <arg name="map_size_x" value="100.0"/>
  <arg name="map_size_y" value="100.0"/>
  <arg name="map_size_z" value=" 10.0"/>

  <arg name="init_x" value="0"/>
  <arg name="init_y" value="0"/>
  <arg name="init_z" value="0"/> 
  <arg name="control_mode" default="feedback"/>  <!-- feedforward / feedback / hybrid -->
  <arg name="max_vel_x" default="2.0"/>  
  <arg name="max_vel_y" default="1.0"/> 
  <arg name="max_vel_z" default="0.6"/>    
  <!-- topic of your odometry such as VIO or LIO -->
  <arg name="odom_topic" value="/odom/ground_truth" />
     <!-- 启动 depth_image_converter 节点 -->
  <node pkg="depth_image_converter"
        type="depth_image_converter_node"
        name="depth_image_converter"
        output="screen">
    <param name="depth_scaling_factor" value="1000.0" />
  </node>

  <node pkg="legged_config" name="position_to_velocity_controller" 
  type="position_to_velocity_controller.py" 
  output="screen">
      <rosparam command="load" file="$(find legged_config)/config/position_controller.yaml"/>

      <param name="odom_topic" value="$(arg odom_topic)"/>
      <param name="position_cmd_topic" value="/planning/pos_cmd"/>
      <param name="cmd_vel_topic" value="/cmd_vel"/>

      <param name="control_mode" value="$(arg control_mode)"/>
      <param name="max_linear_vel_x" value="$(arg max_vel_x)"/>
      <param name="max_linear_vel_y" value="$(arg max_vel_y)"/>
      <param name="max_angular_vel_z" value="$(arg max_vel_z)"/>
</node>
  <!-- main algorithm params -->
  <include file="$(find exploration_manager)/launch/algorithm.xml">

    <arg name="map_size_x_" value="$(arg map_size_x)"/>
    <arg name="map_size_y_" value="$(arg map_size_y)"/>
    <arg name="map_size_z_" value="$(arg map_size_z)"/>

    <arg name="box_min_x" value="-40.0"/>
    <arg name="box_min_y" value="-20.0"/>
    <arg name="box_min_z" value=" 0.0"/>
    <arg name="box_max_x" value="40.0"/>
    <arg name="box_max_y" value="20.0"/>
    <arg name="box_max_z" value=" 2.0"/>

    <arg name="odometry_topic" value="$(arg odom_topic)"/>
    <!-- sensor pose: transform of camera frame in the world frame -->
    <arg name="sensor_pose_topic" value="$(arg odom_topic)"/>
    <!-- depth topic: depth image, 640x480 by default -->
    <!-- cloud topic: point cloud measurement  -->
    <arg name="depth_topic" value="/depth_image/converted"/>
    <arg name="cloud_topic" value="nouse"/>
    <!-- intrinsic params of the depth camera -->
    <arg name="cx" value="321.04638671875"/>
    <arg name="cy" value="243.44969177246094"/>
    <arg name="fx" value="387.229248046875"/>
    <arg name="fy" value="387.229248046875"/>

    <!-- maximum velocity and acceleration the drone will reach -->
    <arg name="max_vel" value="0.6" />
    <arg name="max_acc" value="0.3" />
  </include>

  <!-- trajectory server -->
  <node pkg="plan_manage" name="traj_server" type="traj_server" output="screen">
    <remap from="/position_cmd" to="planning/pos_cmd"/>

    <remap from="/odom_world" to="$(arg odom_topic)"/>
    <param name="traj_server/time_forward" value="1.5" type="double"/>
    <param name="traj_server/pub_traj_id" value="4" type="int"/>
    <param name="traj_server/init_x" value="$(arg init_x)" type="double"/>
    <param name="traj_server/init_y" value="$(arg init_y)" type="double"/>
    <param name="traj_server/init_z" value="$(arg init_z)" type="double"/>

    <param name="perception_utils/top_angle" value="0.56125" type="double"/>
    <param name="perception_utils/left_angle" value="0.69222" type="double"/>
    <param name="perception_utils/right_angle" value="0.68901" type="double"/>
    <param name="perception_utils/max_dist" value="9.5" type="double"/>
    <param name="perception_utils/vis_dist" value="1.0" type="double"/>
    

  </node>

  <node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
    <remap from="~odom" to="$(arg odom_topic)"/>        
    <remap from="~goal" to="/move_base_simple/goal"/>
    <remap from="~traj_start_trigger" to="/traj_start_trigger" />
    <param name="waypoint_type" value="point"/>    
  </node>

  <!-- use simulator -->

  <!-- <node pkg ="map_generator" name ="map_pub" type ="map_pub" output = "screen" args="$(find map_generator)/resource/office.pcd"/>     -->

  <!-- <node pkg="local_sensing_node" type="pcl_render_node" name="pcl_render_node" output="screen">
      <rosparam command="load" file="$(find local_sensing_node)/params/camera.yaml" />
      <param name="sensing_horizon"  value="5.0" />
      <param name="sensing_rate"     value="30.0"/>
      <param name="estimation_rate"  value="30.0"/>
      <param name="camera_height_offset" value="$(arg init_z)"/>
      <param name="map/x_size"     value="$(arg map_size_x)"/>
      <param name="map/y_size"     value="$(arg map_size_y)"/>
      <param name="map/z_size"     value="$(arg map_size_z)"/>
      <remap from="~global_map" to="/map_generator/global_cloud"/>
      <remap from="~odometry"   to="$(arg odom_topic)"/>
  </node> -->
  
  <node pkg="tf" type="static_transform_publisher" name="tf_53" args="0 0 0 0 0 0 world navigation 100" />

   <node if="true" 
      name="rviz_visualization" 
      pkg="rviz" 
      type="rviz" 
      output="screen" 
      args="-d $(find legged_config)/config/leg_exploration.rviz -f odom" />
</launch>